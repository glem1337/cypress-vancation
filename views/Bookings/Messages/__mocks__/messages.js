export default [
  {
    id: '53b85e92-5ce8-4f83-91aa-b5a372157e6f',
    createdAt: '2021-10-05T11:46:05.885Z',
    isReadByOwner: false,
    isReadByUser: true,
    creatorId: '653bbbf6-70ff-4adc-8c37-647f7bf447e6',
    messageDocument: null,
    messageImage: null,
    messageText: {
      id: '19896080-022c-40f4-a6e7-c46c65818b1a',
      createdAt: '2021-10-05T11:46:05.889Z',
      messageId: '53b85e92-5ce8-4f83-91aa-b5a372157e6f',
      text: 'hello, dude',
    },
  },
];
