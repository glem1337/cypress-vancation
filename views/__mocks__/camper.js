export default {
  id: 'a18f7338-c2b3-4cee-be4d-579c0a726c1a',
  name: 'test',
  description: 'desc',
  status: 'draft',
  insurance: 'pending',
  place: 'Astoria, New York, New York 11102, United States',
  publicId: '111',
  estimatedEarning: 1200.0,
  vehicleTypeIconUrl: 'test.svg',
  vehicleTypeName: 'Modern Van',
  placeId: 'neighborhood.290416',
  longitude: -73.93030000000002,
  latitude: 40.772,
  glamper: true,
  owner: {
    id: '74a9c587-7b78-4aa5-ba93-ba3d310be127',
    account: {
      createdAt: '2021-07-09T13:39:26.587Z',
      drivingLicenceVerified: false,
      email: 'kateryna.sessa+owner@rubygarage.org',
      emailVerified: false,
      id: '12563773-6e8f-4861-8abd-82a78632c6ac',
      user: {
        avatarUrl: null,
        createdAt: '2021-07-09T13:39:26.590Z',
        firstName: 'Kat',
        id: '77015afb-71e3-4b94-a9cb-69a97dc6b358',
        lastName: 'Kat',
        updatedAt: '2021-07-09T13:39:26.590Z',
      },
    },
    type: 'owner',
    avatarUrl: null,
    averageRating: null,
    businessTitle: null,
    campersCount: 25,
    createdAt: 'Member since July 2021',
    description: 'description',
    email: null,
    idVerified: false,
  },
  specificationDetail: {
    drivetrain: '4x4',
    fuelType: 'electric',
    insideHeight: 'low_top',
    length: '67.89',
    modelNaming: 'Clair Weimann',
    name: 'Katelyn Lowe',
    seats: 1,
    sleeps: 1,
    stateRegistred: 'Arkansas',
    transmission: 'automatice',
    year: 1585,
    mileage: 'Under 50,000',
    whoBuiltCamper: 'Vasya',
  },
  pricingInfo: {
    id: '74a9c587-7b78-4aa5-ba93-ba3d310be127',
  },
  camperRule: {
    id: '74a9c587-7b78-4aa5-ba93-ba3d310be127',
    bookingApprovalPolicy: 'instant_book',
    cancellationPolicy: 'easy_going',
  },
  deliveryInformation: {
    id: '74a9c587-7b78-4aa5-ba93-ba3d310be127',
    pickup: true,
  },
  camperPhotos: [{
    id: '74a9c587-7b78-4aa5-ba93-ba3d310be127',
    photoUrl1100: 'photoUrl1100',
  }],
  insuranceInfo: {
    id: '74a9c587-7b78-4aa5-ba93-ba3d310be127',
  },
  tripFee: {
    id: '74a9c587-7b78-4aa5-ba93-ba3d310be127',
    cleaning: 1,
    dumpingFee: 2,
    flue: 3,
    lateDropOff: 4,
    customFees: [
      {
        id: 'id',
        name: 'name',
        price: 5,
        frequency: 'per_day',
      },
    ],
    tripFeeMileage: {
      limit: true,
      available: 6,
      overage: 7,
    },
    tripFeeGenerator: {
      limit: true,
      available: 8,
      overage: 9,
    },
  },
  blockedPeriods: [],
  pricingPeriods: [],
  externalCalendars: [],
  camperCalendar: {
    id: 'e1db81b1-8ad5-4e88-9c99-f5c9020148eb',
    externalCalendars: [],
    blockedPeriods: [
      {
        id: '56ceb535-322f-42c6-8dcd-a88ad59000f3',
        startDate: '2021-08-05',
        endDate: '2021-08-06',
      },
      {
        id: '17a32ec0-a486-4247-910c-6c0c57ba6ec3',
        startDate: '2021-08-25',
        endDate: '2021-08-25',
      },
      {
        id: '2c7bbaef-6b00-494f-b005-6a762e90c785',
        startDate: '2021-07-28',
        endDate: '2021-07-28',
      },
    ],
    pricingPeriods: [
      {
        id: '8aa92a51-ef00-4dc6-8b84-2820869b9278',
        startDate: '2021-07-28',
        endDate: '2021-07-28',
        price: '33.0',
      },
      {
        id: '248ba38e-e18a-467c-996b-36aa4bd73e85',
        startDate: '2021-07-29',
        endDate: '2021-07-30',
        price: '320.0',
      },
    ],
    customDiscountPeriods: [
      {
        id: 'ffb3f0cc-4513-4fdc-81d0-03587232d9ee',
        startDate: '2021-07-27',
        endDate: '2021-07-28',
        weeklyDiscountPercent: 15,
        monthlyDiscountPercent: 23,
      },
      {
        id: 'a070339d-7058-4769-8e16-814084c6bfe3',
        startDate: '2021-07-28',
        endDate: '2021-07-28',
        weeklyDiscountPercent: 15,
        monthlyDiscountPercent: 23,
      },
      {
        id: 'aba16cba-5873-4273-8172-816b276a2db7',
        startDate: '2021-07-28',
        endDate: '2021-07-28',
        weeklyDiscountPercent: 44,
        monthlyDiscountPercent: 44,
      },
      {
        id: '586016b9-d351-4ca1-ac8a-373e7e3b8832',
        startDate: '2021-07-29',
        endDate: '2021-07-29',
        weeklyDiscountPercent: 15,
        monthlyDiscountPercent: 33,
      },
    ],
  },
  camperAddition: {
    amenityHealthSafetyItems: [
      {
        healthSafety: {
          id: '74a9c587-7b78-4aa5-ba93-ba3d310be127',
          iconUrl: 'path/to/icon/url',
          name: 'name',
        },
      },
    ],
    restrictionRule: {
      allowPets: false,
      smoking: false,
      festivalApproved: false,
    },
    customRestrictionRules: [
      {
        id: '74a9c587-7b78-4aa5-ba93-ba3d310be127',
        active: true,
        name: 'name',
      },
    ],
    travelRestriction: {
      mexico: false,
      canada: true,
      burningMan: false,
    },
    customTravelRestrictions: [
      {
        id: '74a9c587-7b78-4aa5-ba93-ba3d310be127',
        active: true,
        name: 'name',
      },
    ],
    restrictionRoad: {
      fourWheelRoad: false,
      offRoad: false,
      snowAndIceRoad: false,
      dirtryRoad: true,
    },
    customRestrictionRoads: [
      {
        id: '74a9c587-7b78-4aa5-ba93-ba3d310be127',
        active: true,
        name: 'name',
      },
    ],
    camperQuestions: [
      {
        id: '74a9c587-7b78-4aa5-ba93-ba3d310be127',
        text: 'test',
        required: true,
      },
    ],
    customTravelAccessories: [
      {
        active: true,
        description: '',
        id: 'dd477886-ec22-4da9-ba4c-08627f35fb1a',
        maxAmount: '2.0',
        name: 'Bike Rack',
        price: '1.0',
        priceUnit: 'each',
      },
    ],
  },
};
