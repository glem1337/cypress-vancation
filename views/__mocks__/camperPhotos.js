export default [
  {
    id: 1,
    mainPhoto: true,
    photo: 'http://.../uploads/store/camperphoto',
    camper: {
      id: '0aebbaa8-5006-483b-962c-82fe01ed581f',
      status: 'draft',
    },
  },
  {
    id: 2,
    mainPhoto: false,
    photo: 'http://.../uploads/store/camperphoto',
    camper: {
      id: '0aebbaa8-5006-483b-962c-82fe01ed581f',
      status: 'draft',
    },
  },
];
