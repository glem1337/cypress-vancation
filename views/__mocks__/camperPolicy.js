export default {
  booking_approval_policy: 'instant_book',
  cancellation_policy: 'easy_going',
  camper_id: 'f7445371-9f19-4c52-b0a4-711785cb55f1',
};
