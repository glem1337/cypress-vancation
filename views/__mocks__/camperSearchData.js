import getCamperDataForSearch from 'utils/destinations/getCamperDataForSearch';

import mockedCamper from './camper';

export default getCamperDataForSearch(mockedCamper);
