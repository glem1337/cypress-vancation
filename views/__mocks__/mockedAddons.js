export default [
  {
    active: false,
    description: '',
    iconUrl:
      'https://vancation-api-staging-assets.s3.us-west-1.amazonaws.com/store/travelaccessory/dd477886-ec22-4da9-ba4c-08627f35fb1a/icon/5961aeeef11b1313f39bf3ec1f0f83ed.svg',
    id: 'dd477886-ec22-4da9-ba4c-08627f35fb1a',
    maxAmount: null,
    name: 'Bike Rack',
    price: null,
    priceUnit: 'each',
  },
];
