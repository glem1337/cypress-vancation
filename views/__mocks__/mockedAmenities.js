export default [
  {
    id: '480d527e-a832-4abc-9b6b-5c5b42850dc6',
    title: 'Power System',
    iconUrl: null,
    configurationSubAmenities: [],
    configurationAmenityOptions: [
      {
        id: '730b21ec-2d4a-4ca4-85e8-966388a8c29b',
        type: 'toggler',
        title: 'Off Grid Capable System',
        state: false,
        tooltip: null,
        iconUrl: null,
        configurationSubAmenities: [],
      },
      {
        id: '1973200d-78b3-422c-9020-214189dbe826',
        type: 'checkbox',
        title: 'Electric Power System',
        state: false,
        tooltip: null,
        iconUrl: null,
        configurationSubAmenities: [
          {
            id: 'aafabb27-5065-4240-b99f-2baea96a535f',
            title: 'Battery Storage(under 300 amp hours)',
            tooltip: null,
            iconUrl: null,
            state: false,
          },
          {
            id: 'c9a7626e-2f6b-43c5-a2f3-b78b64c87a2a',
            title: 'Battery Storage(300+ amp hours)',
            tooltip: null,
            iconUrl: null,
            state: false,
          },
          {
            id: '4c944b35-faaf-4515-a24f-1df59c1dfdd7',
            title: 'Alternator Charger',
            tooltip: null,
            iconUrl: null,
            state: false,
          },
          {
            id: '353d6e91-6039-4eca-9f86-e7b220f80a08',
            title: 'Inverter',
            tooltip: null,
            iconUrl: null,
            state: false,
          },
          {
            id: '5df5ee66-7b49-4e94-a440-277f602e0911',
            title: 'A/C Electrical Outlets',
            tooltip: null,
            iconUrl: null,
            state: false,
          },
          {
            id: 'a8d4c375-b91a-467f-9f9e-351fd83e0f6b',
            title: 'Solar Panels',
            tooltip: null,
            iconUrl: null,
            state: false,
          },
        ],
      },
      {
        id: 'aa38a2e0-3d12-435c-a56f-d59fcca76114',
        type: 'checkbox',
        title: 'Propan Fuel System',
        state: false,
        tooltip: null,
        iconUrl: null,
        configurationSubAmenities: [],
      },
      {
        id: 'a02d7663-61c5-42bb-86e4-348f52c73f09',
        type: 'checkbox',
        title: 'Generator',
        state: false,
        tooltip: null,
        iconUrl: null,
        configurationSubAmenities: [],
      },
      {
        id: '6c496237-3ef2-462b-ae3b-f2f4445ef666',
        type: 'toggler',
        title: 'Shore Power Hookups',
        state: false,
        tooltip: null,
        iconUrl: null,
        configurationSubAmenities: [],
      },
    ],
  },
  {
    id: 'a5f1e329-fa2a-4e91-b678-ccca813eb154',
    title: 'Bedroom',
    iconUrl: null,
    configurationSubAmenities: [
      {
        id: 'bf00cdfb-c28a-4065-b32d-dd6e7a3b6e4f',
        title: 'Large Mattress',
        tooltip: null,
        iconUrl: null,
        quantity: 0,
        maxQuantity: 0,
        state: false,
      },
      {
        id: 'b3f036b6-8995-427a-a334-15fd5d20f49e',
        title: 'Extra Storage',
        tooltip: null,
        iconUrl: null,
        quantity: 0,
        maxQuantity: 0,
        state: false,
      },
      {
        id: 'cd0e12e7-9d33-440e-b7be-f12d7fd0f97a',
        title: 'Blackout Shades',
        tooltip: null,
        iconUrl: null,
        quantity: 0,
        maxQuantity: 0,
        state: false,
      },
      {
        id: '9de56b33-8136-47af-8d0e-5c3e283b2368',
        title: 'Sleep Sideways',
        tooltip: null,
        iconUrl: null,
        quantity: 0,
        maxQuantity: 0,
        state: false,
      },
    ],
    configurationAmenityOptions: [],
  },
  {
    id: '8d08c5df-43bc-4597-938e-e18752a9664b',
    title: 'Entertainment',
    iconUrl: null,
    configurationSubAmenities: [
      {
        id: '0925d8a2-3f91-4b2a-a33c-8c1971486a19',
        title: 'Upgraded Speaker System',
        tooltip: null,
        iconUrl: null,
        quantity: 0,
        maxQuantity: 0,
        state: false,
      },
      {
        id: '240b545f-6861-4e0f-b0ae-2d887f81f6e8',
        title: 'Bluetooth Speaker',
        tooltip: null,
        iconUrl: null,
        quantity: 0,
        maxQuantity: 0,
        state: false,
      },
      {
        id: 'ce7a00ef-a5d9-4f01-87f6-74da13b2e797',
        title: 'TV',
        tooltip: null,
        iconUrl: null,
        quantity: 0,
        maxQuantity: 0,
        state: false,
      },
      {
        id: 'd7c5501b-6484-4137-8e4c-9f63f4b63f4e',
        title: 'Wifi Router',
        tooltip: null,
        iconUrl: null,
        quantity: 0,
        maxQuantity: 0,
        state: false,
      },
      {
        id: '4d3893f3-932b-4401-b627-58774ae30bd1',
        title: 'Cell Signal Booster',
        tooltip: null,
        iconUrl: null,
        quantity: 0,
        maxQuantity: 0,
        state: false,
      },
      {
        id: '70093afe-43ba-4706-89d1-23117632fa57',
        title: 'Radio',
        tooltip: null,
        iconUrl: null,
        quantity: 0,
        maxQuantity: 0,
        state: false,
      },
    ],
    configurationAmenityOptions: [],
  },
  {
    id: '93ed1cf3-a645-4912-9d74-3000e67fe3d9',
    title: 'Accomodation',
    iconUrl: null,
    configurationSubAmenities: [
      {
        id: '54cac3fb-2743-4953-b21b-225a2ea47447',
        title: 'King bed',
        tooltip: null,
        iconUrl: null,
        quantity: 10,
        maxQuantity: 10,
        state: false,
      },
      {
        id: 'b55ecf58-6a95-4bf3-b1fe-b8ac1bf1d2a0',
        title: 'Twin bed',
        tooltip: null,
        iconUrl: null,
        quantity: 10,
        maxQuantity: 10,
        state: false,
      },
      {
        id: '8fd7b975-daa4-4a08-87b2-479a53f789e6',
        title: 'Queen bed',
        tooltip: null,
        iconUrl: null,
        quantity: 10,
        maxQuantity: 10,
        state: false,
      },
      {
        id: '9152fe3c-456c-48bc-8b44-af28c11ae584',
        title: 'Single bed',
        tooltip: null,
        iconUrl: null,
        quantity: 10,
        maxQuantity: 10,
        state: false,
      },
      {
        id: 'ef65e471-a83c-46fb-9b0a-6e8583cf21b4',
        title: 'Bunk bed',
        tooltip: null,
        iconUrl: null,
        quantity: 10,
        maxQuantity: 10,
        state: false,
      },
      {
        id: '156e80ca-432f-4850-aba7-e1abcbdc9b91',
        title: 'Dinette conversion',
        tooltip: null,
        iconUrl: null,
        quantity: 10,
        maxQuantity: 10,
        state: false,
      },
    ],
    configurationAmenityOptions: [],
  },
  {
    id: '245850cd-75b6-442f-a18b-837cd91ea6e5',
    title: 'Kitchen',
    iconUrl: null,
    configurationSubAmenities: [
      {
        id: '058785da-2eed-4c91-bac7-4791f54d0686',
        title: 'King bed',
        tooltip: null,
        iconUrl: null,
        quantity: 0,
        maxQuantity: 0,
        state: false,
      },
      {
        id: 'af80fb05-44ea-40ed-ac45-3765cedec993',
        title: 'Twin bed',
        tooltip: null,
        iconUrl: null,
        quantity: 0,
        maxQuantity: 0,
        state: false,
      },
      {
        id: 'a78eb724-4bec-488d-a00e-8afe3997ce0a',
        title: 'Queen bed',
        tooltip: null,
        iconUrl: null,
        quantity: 0,
        maxQuantity: 0,
        state: false,
      },
      {
        id: 'bd41dba4-e7e1-4e17-bc54-15ecfeeb0e87',
        title: 'Single bed',
        tooltip: null,
        iconUrl: null,
        quantity: 0,
        maxQuantity: 0,
        state: false,
      },
      {
        id: '2faad202-0c67-49fc-a785-2e1538666ba0',
        title: 'Bunk bed',
        tooltip: null,
        iconUrl: null,
        quantity: 0,
        maxQuantity: 0,
        state: false,
      },
      {
        id: '915089e7-48a1-4dc3-91f3-b7d82d465944',
        title: 'Dinette conversion',
        tooltip: null,
        iconUrl: null,
        quantity: 0,
        maxQuantity: 0,
        state: false,
      },
    ],
    configurationAmenityOptions: [],
  },
  {
    id: 'ff75edbf-4f60-4b01-b3a0-79fbe2df8c10',
    title: 'Bathroom',
    iconUrl: null,
    configurationSubAmenities: [
      {
        id: '2c826f37-91dc-4482-a67d-d2f2967b97c5',
        title: 'King bed',
        tooltip: null,
        iconUrl: null,
        quantity: 0,
        maxQuantity: 0,
        state: false,
      },
      {
        id: '8be320dd-df33-4225-b8c4-6cb17c65e902',
        title: 'Twin bed',
        tooltip: null,
        iconUrl: null,
        quantity: 0,
        maxQuantity: 0,
        state: false,
      },
      {
        id: 'de3c1f60-ea5d-4b8e-8964-d506c7530bfe',
        title: 'Queen bed',
        tooltip: null,
        iconUrl: null,
        quantity: 0,
        maxQuantity: 0,
        state: false,
      },
      {
        id: 'b4894dcd-eb51-4953-b9fc-6b7142f95237',
        title: 'Single bed',
        tooltip: null,
        iconUrl: null,
        quantity: 0,
        maxQuantity: 0,
        state: false,
      },
      {
        id: 'd9c7b20c-dec8-47f7-bf98-298a696f6844',
        title: 'Bunk bed',
        tooltip: null,
        iconUrl: null,
        quantity: 0,
        maxQuantity: 0,
        state: false,
      },
      {
        id: 'e544a8a3-3bc6-429c-8bdd-f3a26f9665dc',
        title: 'Dinette conversion',
        tooltip: null,
        iconUrl: null,
        quantity: 0,
        maxQuantity: 0,
        state: false,
      },
    ],
    configurationAmenityOptions: [],
  },
  {
    id: 'eab0fdca-16c3-4184-aef8-311b1bf2ed01',
    title: 'Climate Control',
    iconUrl: null,
    configurationSubAmenities: [
      {
        id: '996b052a-e8af-40ce-8749-cd48e906baaf',
        title: 'Ceiling Fan',
        tooltip: null,
        iconUrl: null,
        quantity: 0,
        maxQuantity: 0,
        state: false,
      },
    ],
    configurationAmenityOptions: [
      {
        id: '38b30eae-7d94-4cc7-b08c-44409edece02',
        type: 'toggler',
        title: 'Air conditioner',
        state: false,
        tooltip: null,
        iconUrl: null,
        configurationSubAmenities: [
          {
            id: '8b5acb66-7cf1-4a6b-80ff-0c93d6859ae5',
            title: 'Hookups Required For A/C',
            tooltip: null,
            iconUrl: null,
            state: false,
          },
          {
            id: '3e047a0a-d37b-4d2f-8bf1-d5270b89090b',
            title: 'Off Grid Air Conditioner',
            tooltip: null,
            iconUrl: null,
            state: false,
          },
          {
            id: 'a2226dd2-f731-42d5-b161-4a3179dbaed7',
            title: 'A/C Only While Camper is On',
            tooltip: null,
            iconUrl: null,
            state: false,
          },
        ],
      },
      {
        id: '4bf65e35-d340-486b-b498-1bb31bfa121d',
        type: 'toggler',
        title: 'Heating',
        state: false,
        tooltip: null,
        iconUrl: null,
        configurationSubAmenities: [
          {
            id: '1eb933d8-ac20-4627-bbb8-d362261d03d1',
            title: 'Portable Heater',
            tooltip: null,
            iconUrl: null,
            state: false,
          },
          {
            id: '65418756-c3a6-4f5e-999a-9ffe62494536',
            title: 'Air Heating System',
            tooltip: null,
            iconUrl: null,
            state: false,
          },
          {
            id: 'a4be548a-7f9b-4b8c-89d0-3c57cdfd7120',
            title: 'Heating Only While Camper is On',
            tooltip: null,
            iconUrl: null,
            state: false,
          },
        ],
      },
    ],
  },
];
