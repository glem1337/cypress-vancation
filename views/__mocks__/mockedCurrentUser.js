export default {
  id: '190ddf41-b281-40c0-b48f-7ee98caeacdc',
  email: 'test@com.my',
  createdAt: '2021-07-10T14:47:39.577Z',
  emailVerified: false,
  user: {
    id: '653bbbf6-70ff-4adc-8c37-647f7bf447e6',
    firstName: 'Test',
    lastName: 'Test',
    avatarUrl: null,
    createdAt: '2021-07-10T14:47:39.585Z',
    updatedAt: '2021-07-10T14:47:39.585Z',
  },
};
