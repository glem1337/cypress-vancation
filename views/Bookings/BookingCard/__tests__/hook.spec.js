import { act, renderHook } from '@testing-library/react-hooks';

import {
  BOOKING_STATUS_DATA,
  BOOKING_STATUS,
  BOOKING_STATUS_ERROR,
} from 'constants/booking';
import { camperInquirySelector } from 'state/concepts/booking/selectors';
import { setActiveBookingId as setActiveBookingIdAction } from 'state/concepts/booking/actions';
import { dispatch } from '__mocks__/react-redux';

import useContainer from '../hook';

const mockedInquiry = {
  lastMessageText: 'lastMessageText',
  unreadMessagesAmount: 3,
  booking: null,
  startDate: new Date(),
  endDate: new Date(),
};
jest.mock('state/concepts/booking/selectors', () => ({
  camperInquirySelector: jest.fn(() => mockedInquiry),
}));

describe('BookingCard useContainer hook', () => {
  let result = null;

  const props = {
    id: 33,
    isFirst: true,
  };

  beforeEach(() => {
    ({ result } = renderHook(() => useContainer(props)));

    jest.clearAllMocks();
  });

  it('matches snapshot', () => {
    expect(result.current).toMatchSnapshot();
  });

  describe('checks `detectStatus` method', () => {
    it('should return inquiry', () => {
      let res = null;

      act(() => {
        res = result.current.detectStatus();
      });

      expect(res).toEqual(BOOKING_STATUS_DATA[BOOKING_STATUS.INQUIRY]);
    });

    it('should return pending', () => {
      let res = null;

      camperInquirySelector.mockReturnValueOnce({
        ...mockedInquiry,
        booking: {
          status: BOOKING_STATUS.PENDING,
        },
      });

      ({ result } = renderHook(() => useContainer(props)));

      act(() => {
        res = result.current.detectStatus();
      });

      expect(res).toEqual(BOOKING_STATUS_DATA[BOOKING_STATUS.PENDING]);
    });

    it('should return error', () => {
      let res = null;

      camperInquirySelector.mockReturnValueOnce({
        ...mockedInquiry,
        booking: {
          status: 'test',
        },
      });

      ({ result } = renderHook(() => useContainer(props)));

      act(() => {
        res = result.current.detectStatus();
      });

      expect(res).toEqual(BOOKING_STATUS_ERROR);
    });
  });

  describe('checks `constructDatesString` method', () => {
    it('should return empty string', () => {
      let res = null;

      camperInquirySelector.mockReturnValueOnce(null);

      ({ result } = renderHook(() => useContainer(props)));

      act(() => {
        res = result.current.constructDatesString();
      });

      expect(res).toBe('');
    });

    it('should return date string', () => {
      let res = null;

      camperInquirySelector.mockReturnValueOnce({
        ...mockedInquiry,
        booking: {
          status: 'test',
        },
      });

      ({ result } = renderHook(() => useContainer(props)));

      act(() => {
        res = result.current.constructDatesString();
      });

      expect(res).toBe('Feb 20 - Feb 20, 2000');
    });
  });

  it('checks `setActiveBookingId` method', () => {
    act(() => {
      result.current.setActiveBookingId();
    });

    expect(dispatch).toHaveBeenCalledWith(setActiveBookingIdAction(props.id));
  });
});
