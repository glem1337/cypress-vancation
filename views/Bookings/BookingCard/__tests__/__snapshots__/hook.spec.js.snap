// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BookingCard useContainer hook matches snapshot 1`] = `
Object {
  "camperName": undefined,
  "constructDatesString": [Function],
  "datesString": "Feb 20 - Feb 20, 2000",
  "detectStatus": [Function],
  "id": 33,
  "isFirst": true,
  "lastMessage": "lastMessageText",
  "lastMessageDate": "2 days ago",
  "ownerInfo": Object {
    "avatar": "https://randomuser.me/api/portraits/women/87.jpg",
    "isOnline": false,
    "name": "Dean Joseph",
  },
  "setActiveBookingId": [Function],
  "status": Object {
    "color": "blue",
    "id": "shared.inquiry",
  },
  "unreadMessagesAmount": 3,
}
`;
